### PR TITLE
Add exception handling for book modification 

### LINKED ISSUE #758: Error by opening a file is not handled properly
Steps:
* Have a file in the local library
* Delete or rename that file in the filesystem
* Click "open" on that file in the local library

I get:
* A crash with:
```
$ ./kiwix-desktop
Compiled with Qt Version  5.12.8
Runtime Qt Version  5.12.8
add widget
terminate called after throwing an instance of 'std::runtime_error'
  what():  error 2 opening file "/home/kelson/Downloads/5.zim
Aborted (core dumped)
```

I would like:
* A proper error dialog with an error... and probably the entry should be deleted from the local library.

### PR DESCRIPTION
**Pull request by juuz0**
Fixes #758

### CODE REVIEW

**Comment by juuz0** [comment]
Have encountered a small bug, fixing

File: resources/i18n/en.json [hunk 1]
```diff
@@ -121,5 +121,6 @@
     "open-link-in-web-browser":"Open link in web browser",
     "download-dir-dialog-title":"Are you sure you want to change the download directory?",
     "download-dir-dialog-msg":"The new download directory path will be:\n{{DIRECTORY}}",
-    "invalid-port":"Invalid port"
+    "invalid-port":"Invalid port",
+    "zim-open-fail":"The zim file has been modified on disk and cannot be loaded properly.\nPlese undo the changes and re-open it."
```
Comment (kelson42) [inline resources/i18n/en.json:125]: "The ZIM file %1 cannot be open properly. It will be removed from your library."

**Review by kelson42 — CHANGES_REQUESTED** [review]
@juuz0 Beside changing the message, it looks good to me. Thx!

**Comment by kelson42** [comment]
@juuz0 I wonder why starting `/kiwix-desktop file_which_does_not_exist.zim` does not generate the same error? Is there a good reason? Could we have the same behaviour?

File: resources/i18n/en.json (hunk 1 above)
Comment (juuz0) [reply resources/i18n/en.json:125]: What should be shown at %1? ZIM title or full path?

**Comment by juuz0** [comment]
> @juuz0 I wonder why starting /kiwix-desktop file_which_does_not_exist.zim does not generate the same error? Is there a good reason? Could we have the same behaviour
Yes we should do that but the possibilities here are a bit different and so different message (file may/may not be in library) should be displayed
`kiwix-desktop justSomeRandomWords`
`kiwix-desktop aValidZimFile.zim`
`kiwix-desktop anInvalidZimFile.zim`

What do you think about this?

File: resources/i18n/en.json (hunk 1 above)
Comment (kelson42) [reply resources/i18n/en.json:125]: fullpath

**Comment by juuz0** [comment]
>  I wonder why starting /kiwix-desktop file_which_does_not_exist.zim does not generate the same error? Is there a good reason? Could we have the same behaviour?

I have found that this is already implemented except missing a `throw` statement

**Review by kelson42 — CHANGES_REQUESTED** [review]
It I launch with a non existing file from the command line I get something like this:
image

This is not the same box style like if the error is displayed from library and there is a checkbox "Show this message again which should not be there".

**Comment by juuz0** [comment]
Consistent now, although I don't really like this copying almost the same code for MessageBox again (but its twice only so maybe fine!) 
Maybe re-implementing `KiwixApp::showMessage()` (currently uses default message box hence the "show this message again") is better...Would go with your inputs

File: src/tabbar.cpp [hunk 2]
```diff
@@ -154,7 +154,11 @@ void TabBar::openUrl(const QUrl& url, bool newTab)
         webView = createNewTab(true, true)->getWebView();
     }
     QUITIFNULL(webView);
-    webView->setUrl(url);
+    try {
+        webView->setUrl(url);
+    } catch (const std::exception& e) {
+        throw;
+    }
```
[the remark this answers is not in the document]
Comment (juuz0) [reply src/tabbar.cpp:161]: I had some more code there, removed later and forgot to remove this. Thanks! :)

File: src/library.cpp [hunk 3]
```diff
@@ -48,6 +48,9 @@ QString Library::openBookFromPath(const QString &zimPath)
 
     kiwix::Manager manager(&m_library);
     auto id =  manager.addBookFromPathAndGetId(zimPath.toStdString());
+    if (id == "") {
+        throw std::invalid_argument("invalid zim file");
+    }
```
[the remark this answers is not in the document]
Comment (juuz0) [reply src/library.cpp:53]: Agreed, main cause of all this was `addBookFromPathAndGetId` not throwing expected exception.

**Review by kelson42 — APPROVED** [review]

**Review by mgautierfr — APPROVED** [review]

**Pull request merged**